### PR TITLE
media-gfx/digikam: 5.3.0 version bump

### PR DESCRIPTION
**Pull request by a17r**
@gentoo/kde @akhuettel

### CODE REVIEW

File: media-gfx/digikam/digikam-5.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+if [[ ${KDE_BUILD_TYPE} != live ]]; then
+	KDE_HANDBOOK="true"
+	KDE_TEST="true"
+fi
+CMAKE_MAKEFILE_GENERATOR="emake"
```
Comment (SoapGentoo) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: do you really need to specify this?

File: media-gfx/digikam/digikam-5.3.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	$(add_qt_dep qtwidgets)
+	$(add_qt_dep qtxml)
+	dev-libs/expat
+	media-gfx/exiv2:=
+	media-libs/lcms:2
+	media-libs/liblqr
+	>=media-libs/libpgf-6.12.27
+	media-libs/libpng:0=
+	media-libs/opencv:=[-qt4]
+	|| ( <media-libs/opencv-3.0.0 >=media-libs/opencv-3.1.0 )
```
Comment (SoapGentoo) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: `|| ()` is a touchy topic. The rule is that "most preferred" should always be the first item in the list, in order to ease the problem of satisfying dependency resolution. You sure about this? https://archives.gentoo.org/gentoo-dev/message/6977c4c839e3ff3356d159cbe0c173cd

File: media-gfx/digikam/digikam-5.3.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	media-libs/tiff:0
+	virtual/jpeg:0
+	addressbook? (
+		$(add_kdeapps_dep akonadi-contacts)
+		$(add_kdeapps_dep kcontacts)
+	)
+	calendar? ( $(add_kdeapps_dep kcalcore) )
+	gphoto2? ( media-libs/libgphoto2:= )
+	jpeg2k? ( media-libs/jasper )
```
Comment (SoapGentoo) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: needs `:=` nowadays

File: media-gfx/digikam/digikam-5.3.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	video? ( $(add_qt_dep qtmultimedia 'widgets') )
+	X? (
+		$(add_qt_dep qtx11extras)
+		x11-libs/libX11
+	)
+"
+DEPEND="${COMMON_DEPEND}
+	$(add_qt_dep qtconcurrent)
+	dev-cpp/eigen:3
+	dev-libs/boost[threads]
```
Comment (SoapGentoo) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: This seems somewhat contradictory. So it somehow needs `threads` (which are implemented with a shared object shim) but cannot be combined with `:=`, as it is in `DEPEND`s. Could you check whether the package links against `/usr/lib64/libboost_thread.so`?

File: media-gfx/digikam/digikam-5.3.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+"
+
+RESTRICT=test
+# bug 366505
+
+# FIXME: Unbundle libraw (libs/rawengine/libraw)
+pkg_pretend() {
+	if use openmp ; then
+		tc-has-openmp || die "Please switch to an openmp compatible compiler"
+	fi
```
Comment (SoapGentoo) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: the modern OpenMP checking idiom is: `use openmp && tc-check-openmp` (which presents a suitable error message). Also, this generally needs a `MERGE_TYPE` check too. Please also repeat the check in `pkg_setup`, as `pkg_pretend` cannot be relied upon for checking invariants.

File: media-gfx/digikam/digikam-5.3.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+src_configure() {
+	# LQR = only allows to choose between bundled/external
+	local mycmakeargs=(
+		-DENABLE_APPSTYLES=ON
+		-DENABLE_AKONADICONTACTSUPPORT=$(usex addressbook)
+		-DENABLE_MYSQLSUPPORT=$(usex mysql)
+		-DENABLE_MEDIAPLAYER=$(usex video)
+		-DENABLE_OPENCV3=$(has_version ">=media-libs/opencv-3" && echo yes || echo no)
```
Comment (SoapGentoo) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: This seems brittle. compile-time properties depend on an `RDEPEND` that can in theory be replaced later on. Is it such a problem just hard depending on OpenCV 3 and later?

File: media-gfx/digikam/digikam-5.3.0.ebuild (hunk 4 above)
Comment (a17r) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: It does not, from what I can see.

File: media-gfx/digikam/digikam-5.3.0.ebuild (hunk 1 above)
Comment (a17r) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: https://bugs.gentoo.org/show_bug.cgi?id=588538

File: media-gfx/digikam/digikam-5.3.0.ebuild (hunk 2 above)
Comment (a17r) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: The order is correct in this case. digikam-5.3.0 will be stabilised long before any version of opencv-3 is going to be.

File: media-gfx/digikam/digikam-5.3.0.ebuild (hunk 6 above)
Comment (a17r) [inline media-gfx/digikam/digikam-5.3.0.ebuild]: works together with opencv:= RDEPEND

**Comment by johu** [comment]
Thanks, picked.

**Pull request closed without merging**